ts/screens/profile/ServicesPreferenceScreen.tsx

```diff
@@ -1,75 +1,110 @@
 import { ContentWrapper, VSpacer } from "@pagopa/io-app-design-system";
 import * as pot from "@pagopa/ts-commons/lib/pot";
-import * as React from "react";
-import { connect } from "react-redux";
+import React, { ReactElement, useCallback, useEffect } from "react";
+import { useStore } from "react-redux";
 import { ServicesPreferencesModeEnum } from "../../../definitions/backend/ServicesPreferencesMode";
 import { RNavScreenWithLargeHeader } from "../../components/ui/RNavScreenWithLargeHeader";
 import I18n from "../../i18n";
 import { profileUpsert } from "../../store/actions/profile";
-import { Dispatch } from "../../store/actions/types";
 import {
   profileSelector,
   profileServicePreferencesModeSelector
 } from "../../store/reducers/profile";
 import { GlobalState } from "../../store/reducers/types";
 import { getFlowType } from "../../utils/analytics";
 import { useOnFirstRender } from "../../utils/hooks/useOnFirstRender";
-import { showToast } from "../../utils/showToast";
 import LoadingSpinnerOverlay from "../../components/LoadingSpinnerOverlay";
+import { useIODispatch, useIOSelector } from "../../store/hooks";
+import { IOToast } from "../../components/Toast";
+import { usePrevious } from "../../utils/hooks/usePrevious";
 import {
   trackServiceConfiguration,
   trackServiceConfigurationScreen
 } from "./analytics";
 import { useManualConfigBottomSheet } from "./components/services/ManualConfigBottomSheet";
 import ServicesContactComponent from "./components/services/ServicesContactComponent";
 
-type Props = ReturnType<typeof mapStateToProps> &
-  ReturnType<typeof mapDispatchToProps>;
-
 /**
  * Display the current profile services preference mode (auto or manual)
  * User can update his/her mode
  * @param props
  * @constructor
  */
-const ServicesPreferenceScreen = (props: Props): React.ReactElement => {
-  const { potProfile, state } = props;
+const ServicesPreferenceScreen = (): ReactElement => {
+  const store = useStore();
+  const state = store.getState();
+  const dispatch = useIODispatch();
+  const profile = useIOSelector(profileSelector);
+  const prevProfile = usePrevious(profile);
+  const isLoading = pot.isUpdating(profile) || pot.isLoading(profile);
+  const profileServicePreferenceMode = useIOSelector(
+    profileServicePreferencesModeSelector
+  );
+
+  const dispatchServicePreferencesSetting = useCallback(
+    (mode: ServicesPreferencesModeEnum) =>
+      dispatch(
+        profileUpsert.request({ service_preferences_settings: { mode } })
+      ),
+    [dispatch]
+  );
+
+  const onServicePreferenceSelected = (
```

Comment:
What about using a callback? What do you think?

Reply:
done!